lint/linter.py

```diff
@@ -559,12 +558,12 @@ def assign(cls, view, linter_name=None, reset=False):
         vid = view.id()
         persist.views[vid] = view
         syntax = util.get_syntax(view)
+        persist.debug("detected syntax: {}".format(syntax))
 
-        if not syntax:
-            cls.remove(vid)
+        if not syntax:  # seems like a very rare, edge case
```

Comment:
Is it possible that a view has no syntax?

Reply:
I don't think so. `get_syntax` is rigged to return an empty string in some cases, but I don't think Sublime Text has views without a syntax. Perhaps you can use the syntax map to map to `''` as a syntax to make this happen? Seems like a weird edge case anyway. Perhaps it's a kind of defensiveness that made sense during refactoring this, but no longer.